clients/spark/core/src/test/resources/parser-test/sst_files_generator.go

```diff
@@ -0,0 +1,291 @@
+	FuzzerSeed            = 50
+	// DefaultCommittedPermanentMaxRangeSizeBytes - the max file size that can be written by graveler (pkg/config/config.go)
+	DefaultCommittedPermanentMaxRangeSizeBytes = 20 * MiBToBytes
+	DefaultSstSizeBytes                        = 50 * KiBToBytes
+	DefaultTwoLevelSstSizeBytes                = 10 * KiBToBytes
+	DefaultMaxSizeKb                           = 100
+)
+
+var DefaultUserProperties = map[string]string{
+	"user_prop_1": "val1",
+	"user_prop_2": "val2",
+}
+
+type Entry struct {
+	Key   string
+	Value string
+}
+
+func main() {
+	writeTwoLevelIdxSst()
+	writeMultiSizedSstsWithContentsFuzzing()
+	writeSstsWithWriterOptionsFuzzing()
+	writeSstsWithUnsupportedWriterOptions()
+	writeLargeSsts()
+}
+
+func generateNanoid() (string, error) {
+	return nanoid.New(DefaultValueSizeBytes)
+}
+
+func newGenerateFuzz() func() (string, error) {
+	f := fuzz.NewWithSeed(FuzzerSeed)
+	return func() (string, error) {
+		var val string
+		f.Fuzz(&val)
+		return val, nil
+	}
+}
+
+func writeSstsWithUnsupportedWriterOptions() {
+	sizeBytes := DefaultSstSizeBytes
+	keys := prepareSortedSliceWithNanoid(sizeBytes)
+
+	// Checksum specifies which checksum to use. lakeFS supports parsing sstables with crc checksum.
+	// ChecksumTypeXXHash and ChecksumTypeNone are unsupported and therefore not tested.
+	writerOptions := sstable.WriterOptions{
+		Compression: sstable.SnappyCompression,
+		Checksum:    sstable.ChecksumTypeXXHash64,
+	}
+	createTestInputFiles(keys, generateNanoid, sizeBytes,
+		"checksum.type.xxHash64", writerOptions)
+
+	// TableFormat specifies the format version for writing sstables. The default
+	// is TableFormatRocksDBv2 which creates RocksDB compatible sstables. this is the only format supported by lakeFS.
+	writerOptions = sstable.WriterOptions{
+		Compression: sstable.SnappyCompression,
+		TableFormat: sstable.TableFormatLevelDB,
+	}
+	createTestInputFiles(keys, generateNanoid, sizeBytes,
+		"table.format.leveldb", writerOptions)
+}
+
+func writeSstsWithWriterOptionsFuzzing() {
+	sizeBytes := DefaultSstSizeBytes
+	keys := prepareSortedSliceWithNanoid(sizeBytes)
+
+	// Use fuzzing to define sstable user properties.
+	generateFuzz := newGenerateFuzz()
+	userProps := map[string]string{}
+	const numOfUserDefinedProps = 20
+	for i := 1; i <= numOfUserDefinedProps; i++ {
+		propKey, _ := generateFuzz()
+		propVal, _ := generateFuzz()
+		userProps[propKey] = propVal
+		fmt.Printf("user property %d, key=%s, val=%s\n", i, propKey, propVal)
+	}
+	writerOptions := sstable.WriterOptions{
+		Compression:             sstable.SnappyCompression,
+		TablePropertyCollectors: []func() sstable.TablePropertyCollector{NewStaticCollector(userProps)},
+	}
+	createTestInputFiles(keys, generateNanoid, sizeBytes, "fuzz.table.properties", writerOptions)
+
+	// BlockRestartInterval is the number of keys between restart points
+	// for delta encoding of keys.
+	// The default value is 16.
+	blockRestartInterval := 1 + rand.Intn(100)
```

Comment:
`rand` is still not seeded AFAICT; https://github.com/treeverse/lakeFS/pull/2432#discussion_r700965505 vanished due to a push, but I believe that it remains unresolved.